common/FileSystem.cpp

```diff
@@ -2493,6 +2493,11 @@ bool FileSystem::DeleteDirectory(const char* path)
 std::string FileSystem::GetProgramPath()
 {
 #if defined(__linux__)
+	// Check if we are running inside appimage. If so, return the path to the appimage instead.
+	if (const char* appimage_path = getenv("APPIMAGE"))
+		return std::string(appimage_path);
+
+	// Otherwise, find the executable
```

Comment:
This breaks other usages of FileSystem::GetProgramPath for the app image.

Notably the CI-generated AppImage fails to start because PCSX2 can't find its resources since it tries to look for them relative to the .AppImage file instead of relative to the actual executable.

The crash handler also uses this function, and passes the result to libbacktrace, which I believe uses it to try and load debug info and so wants the real executable path as well.

As a general rule, if you're changing the behaviour of a function like this, search for existing usages of the function e.g using GitHub's search feature or grep to see if your changes will cause problem.